keras_cv/layers/preprocessing/resizing_test.py

```diff
@@ -0,0 +1,246 @@
+                np.ones((2, 2, 1)),
+            ],
+            dtype="float32",
+        )
+        layer = cv_layers.Resizing(
+            2,
+            2,
+            interpolation="nearest",
+            crop_to_aspect_ratio=crop_to_aspect_ratio,
+        )
+        outputs = layer(inputs)
+        expected_output = [
+            [[[1.0], [1.0]], [[1.0], [1.0]]],
+            [[[1.0], [1.0]], [[1.0], [1.0]]],
+            [[[1.0], [1.0]], [[1.0], [1.0]]],
+            [[[1.0], [1.0]], [[1.0], [1.0]]],
+        ]
+        self.assertIsInstance(outputs, tf.Tensor)
+        self.assertNotIsInstance(outputs, tf.RaggedTensor)
+        self.assertAllEqual(expected_output, outputs)
+
+    def test_output_dtypes(self):
+        inputs = np.array([[[1], [2]], [[3], [4]]], dtype="float64")
+        layer = cv_layers.Resizing(2, 2)
+        self.assertAllEqual(layer(inputs).dtype, "float32")
+        layer = cv_layers.Resizing(2, 2, dtype="uint8")
+        self.assertAllEqual(layer(inputs).dtype, "uint8")
+
+    @parameterized.named_parameters(
+        ("batch_crop_to_aspect_ratio", True, False, True),
+        ("batch_dont_crop_to_aspect_ratio", False, False, True),
+        ("single_sample_crop_to_aspect_ratio", True, False, False),
+        ("single_sample_dont_crop_to_aspect_ratio", False, False, False),
+        ("batch_pad_to_aspect_ratio", False, True, True),
+        ("single_sample_pad_to_aspect_ratio", False, True, False),
+    )
+    def test_static_shape_inference(
+        self, crop_to_aspect_ratio, pad_to_aspect_ratio, batch
+    ):
+        channels = 3
+        input_height = 8
+        input_width = 8
+        target_height = 4
+        target_width = 6
+        layer = cv_layers.Resizing(
+            target_height,
+            target_width,
+            crop_to_aspect_ratio=crop_to_aspect_ratio,
+            pad_to_aspect_ratio=pad_to_aspect_ratio,
+        )
+        unit_test = self
+
+        @tf.function
+        def tf_function(img):
+            unit_test.assertListEqual(
+                [input_height, input_width, channels], img.shape.as_list()[-3:]
+            )
+            img = layer(img)
+            unit_test.assertListEqual(
+                [target_height, target_width, channels],
+                img.shape.as_list()[-3:],
+            )
+            return img
+
+        if batch:
+            input_shape = (2, input_height, input_width, channels)
+        else:
+            input_shape = (input_height, input_width, channels)
+        img_data = np.random.random(size=input_shape).astype("float32")
+        tf_function(img_data)
+
+    def test_pad_to_size_with_bounding_boxes_ragged_images(self):
+        images = tf.ragged.constant(
+            [
+                np.ones((8, 8, 3)),
+                np.ones((8, 4, 3)),
+                np.ones((4, 8, 3)),
+                np.ones((2, 2, 3)),
+            ],
+            dtype="float32",
+        )
+        boxes = tf.ragged.stack(
```

Comment:
I think given the complexity of the code, it would be good to have a bounding box test case where we assert the numerics of the resulting boxes

Reply:
Great point; I added some assertions on the numerics here.